builder/vsphere/common/step_boot_command.go

```diff
@@ -153,3 +205,32 @@ func (s *StepBootCommand) Run(ctx context.Context, state multistep.StateBag) mul
 }
 
 func (s *StepBootCommand) Cleanup(_ multistep.StateBag) {}
+
+func hostIP(ifname string) (string, error) {
+	var addrs []net.Addr
+	var err error
+
+	if ifname != "" {
+		iface, err := net.InterfaceByName(ifname)
+		if err != nil {
+			return "", err
+		}
+		addrs, err = iface.Addrs()
+		if err != nil {
+			return "", err
+		}
+	} else {
+		addrs, err = net.InterfaceAddrs()
+		if err != nil {
+			return "", err
+		}
+	}
+	for _, addr := range addrs {
+		if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+			if ipnet.IP.To4() != nil {
```

Comment:
Are we only supporting IPv4 here?

Reply:
Good catch! Resolved in [4f68e62.](https://github.com/hashicorp/packer-plugin-vsphere/commit/4f68e62dea97655f36f46919769bd41c09fffe6d).